sdk/lib/atl/atlcom.h

```diff
@@ -456,7 +456,7 @@ class CComCreator2
 public:
     static HRESULT WINAPI CreateInstance(void *pv, REFIID riid, LPVOID *ppv)
     {
-        ATLASSERT(ppv != NULL && riid != NULL);
+        ATLASSERT(ppv != NULL && riid != (REFIID)NULL);
```

Comment:
How does this work, even with the cast? Wouldn't you have to compare `&riid` to null, since it's a reference?

Reply:
Yeah, when I wrote this I wanted to take into account the case that REFIID can be a reference or a pointer,
but since this is in a templated class that does not make sense anyway.